packages/whenable/src/types.js

```diff
@@ -0,0 +1,43 @@
+export {};
+
+/**
+ * @template T
+ * @typedef {PromiseLike<T | Whenable<T>>} PromiseWhenable
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {import('@endo/pass-style').CopyTagged<
+ *   'Whenable',
+ *   { whenable0: { shorten(): Promise<T | Whenable<T>>} }
+ * >} Whenable
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {{
+ *   whenable: Whenable<T>,
+ *   settler: Settler<T>,
+ * }} WhenableKit
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {{
+ *   whenable: Whenable<T>,
+ *   settler: Settler<T>,
+ *   promise: Promise<T>
+ * }} WhenablePromiseKit
```

Comment:
The PR comment says
`makeWhenablePromiseKit(): { promise, settler: { resolve, reject } }`

Just checking that you only omitted `whenable` from this for expository reasons, yes?

Reply:
Unresolving.

When would you want to use `makeWhenablePromiseKit` rather than `makeWhenableKit`?

Reply:
In the definition of the `when` operator, as above.

Reply:
Assuming we're now agreed to hide `makeWhenablePromiseKit`, resolving. If I misunderstood, please unresolve.